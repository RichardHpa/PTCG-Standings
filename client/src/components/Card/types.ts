import type { ReactNode } from 'react';

export interface CardProps {
  children: ReactNode;
  growHeight?: boolean;
  title?: string;
  action?: ReactNode;
}
