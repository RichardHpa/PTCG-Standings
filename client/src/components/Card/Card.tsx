import clsx from 'clsx';

import type { FC } from 'react';
import type { CardProps } from './types';

export const Card: FC<CardProps> = ({
  growHeight = true,
  children,
  title,
  action,
}) => {
  return (
    <div
      className={clsx(
        'flex flex-col rounded-xl border border-gray-100 bg-white dark:border-gray-700 dark:bg-gray-900 dark:text-gray-400',
        { 'h-full': growHeight },
      )}
    >
      <div className="flex items-center justify-between rounded-t-lg border-b border-gray-200 bg-gray-50 px-3 py-2.5 dark:border-gray-700 dark:bg-gray-700">
        <h2>{title}</h2>
        {action && <div className="flex gap-2">{action}</div>}
      </div>

      {children}
    </div>
  );
};
