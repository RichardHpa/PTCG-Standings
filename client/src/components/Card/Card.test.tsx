import { describe, expect, test } from 'vitest';
import { render, screen } from 'tests/testingLibrary';

import { Card } from './Card';

describe('Card', () => {
  test('renders the component', () => {
    render(<Card title="card title">children</Card>);
    expect(
      screen.getByRole('heading', { level: 2, name: 'card title' }),
    ).toBeInTheDocument();

    expect(screen.queryByRole('button')).not.toBeInTheDocument();
  });

  test('render an action', () => {
    render(
      <Card title="card title" action={<button>action</button>}>
        children
      </Card>,
    );
    expect(screen.getByRole('button', { name: 'action' })).toBeInTheDocument();
  });
});
