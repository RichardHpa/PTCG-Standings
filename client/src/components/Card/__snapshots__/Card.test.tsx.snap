// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="flex flex-col overflow-hidden rounded-xl border border-gray-100 bg-white dark:border-gray-700 dark:bg-gray-900 dark:text-gray-400 h-full"
  >
    <div
      class="flex items-center justify-between rounded-t-lg border-b border-gray-200 bg-gray-50 px-3 py-2.5 dark:border-gray-700 dark:bg-gray-700"
    >
      <h2 />
    </div>
  </div>
</DocumentFragment>
`;
